tests/test_generic.py

```diff
@@ -218,9 +218,11 @@ def test_name_object(caplog):
 
     caplog.clear()
     b = BytesIO()
-    NameObject("hello").write_to_stream(b)
-    assert bytes(b.getbuffer()) == b"hello"
-    assert "Incorrect first char" in caplog.text
+    try:
+        NameObject("hello").write_to_stream(b)
+        pytest.fail("Corrupted name object")
+    except PyPdfError:
+        pass
```

Comment:
This sounds more common

```suggestion
     with pytest.raises(PyPdfError) as exc::
        NameObject("hello").write_to_stream(b)
```

Reply:
I personally even prefer to check the error message as well to avoid side effects, but this probably is a matter of taste.